InnerEye/ML/utils/dataset_util.py

```diff
@@ -212,3 +212,33 @@ def add_label_stats_to_dataframe(input_dataframe: pd.DataFrame,
                 input_dataframe.loc[input_dataframe.subject == subject_id, "channel"].map(col_stats)
 
     return input_dataframe
+
+
+def validated_channel_ids(channels: List[str],
+                          rows: pd.DataFrame,
+                          local_dataset_root_folder: Path,
+                          patient_id: str) -> List[Path]:
+    """
+    Returns the full path for files specified in the training, validation and testing datasets.
+    Ensures that: 1) given any channel type (e.g.,  image_channels or ground_truth_ids or mask_id) defined in the config file
+    there is associated row value for any given subject. 2) exits if there is no corresponding corresponding mapping from
+    subject to channel.  This function is called from InnerEyeContainer->setup (lighting_base.py) and
+    load_dataset_sources->get_paths_for_channel_ids (full_image_dataset.py)
```

Comment:
I'd remove that, such a comment is likely to run out of sync

Reply:
Removed